### PR TITLE
Refactor the params order in Flag init

### PR DESCRIPTION
**Pull request by oarrabi**
Added example entry to command
Adding parent to Command init
Symplifying the generation

@goyox86

### CODE REVIEW

File: Sources/App/add.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+//  remote.swift
+//  Guaka
+//
+//  Created by Omar Abdelhafith on 05/11/2016.
+//
+//
+
+import Guaka
+
+var addCommand: Command = Command(
```
Comment (goyox86) [inline Sources/App/add.swift]: Just a question. Can't we just omit the type here? Just using the inference. Is it a swift idiom to state types like this? #justaskingbro :)

File: Sources/App/add.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  name: "add", 
+  run: executeAddCommand, parent: rootCommand)
```
Comment (goyox86) [inline Sources/App/add.swift]: Just brainfarting here could we also have an API that in the constructor receives a closure? that would be slick.

File: Sources/App/init.swift [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+//  Guaka
+//
+//  Created by Omar Abdelhafith on 05/11/2016.
+//
+//
+
+import Guaka
+
+var initCommand: Command = Command(
+  name: "init", run: executeInitCommand, parent: rootCommand)
```
Comment (goyox86) [inline Sources/App/init.swift]: Same thing about the closure here 😎

File: Sources/App/main.swift [hunk 4]
```diff
@@ -8,9 +8,26 @@
...
-setup()
 
-root.add(flag: Flag(longName: "togge", value: false, shortName: "t", inheritable: false))
-root.add(flag: Flag(longName: "root", value: 1, shortName: "r", inheritable: false))
+[
+  Flag(longName: "foo", value: "", inheritable: true),
+  Flag(longName: "remote", value: true, inheritable: true),
+  Flag(longName: "bar", value: "", inheritable: false),
+  Flag(longName: "xx", value: true, inheritable: false),
+  ].forEach { initCommand.add(flag: $0) }
```
Comment (goyox86) [inline Sources/App/main.swift]: Why aren't we adding this flags to init command on the init command file?

Comment (goyox86) [inline Sources/App/main.swift]: Also why we don't add an 'add' version that just receives an array of flags (or just use it if already have it)?

File: Sources/App/add.swift (hunk 2 above)
Comment (oarrabi) [inline Sources/App/add.swift]: All functions names are closures in swift. so run will work like that, it either accepts a function name, or a closure. (swift is coooul)

File: Sources/App/main.swift (hunk 4 above)
Comment (oarrabi) [inline Sources/App/main.swift]: The init will be generated from the cli, the flags will be mostly added by the dev (that is inline with cobra as it does not have the ability to add flags)

Having said that, the constructor accepts a `[Flag]` so we can add them there!

Comment (oarrabi) [inline Sources/App/main.swift]: Was just lazy here :D

**Comment by goyox86** [comment]
:shipit: 👍

**Pull request merged**